### ISSUE TITLE
Syncthing Security - Information Leakage - Global Discovery

### ISSUE THREAD

**Issue by PascalHeynmoeller** [freedombox/Plinth#1115]
Default Value is = ON
https://docs.syncthing.net/users/security.html
"Global discovery defaults to on.
When turned off, devices with dynamic addresses not on the local network cannot be found and connected to.
An eavesdropper on the Internet can deduce which machines are running Syncthing with global discovery enabled, and what their device IDs are.
The operator of the discovery server can map arbitrary device addresses to IP addresses, and deduce which devices are connected to each other."

Default configuration in Freedombox should be 'Global Discovery = OFF'.

**Comment by JosephKiranBabu** [comment]
@PascalHeynmoeller Thanks for pointing this out. Global discovery might be unnecessary for most home users.
But it might be desired in some cases, like a distributed family that wants to share photos with a family member whose device is not physically available. I think we should mention this in the manual and how to temporarily enable Global Discovery if required. I agree that Global Discovery should be turned OFF by default.

**Comment by PascalHeynmoeller** [comment]
Thx @JosephKiranBabu for the fast commit of the change.

It would be better for anonymity to provide access to Syncthing only as TOR hidden service from the outside to prevent leakage. As mentioned in the Synthing manual proxy security section.
Else we can't provide anonymity.
This is maybe also  a new issue like relaying.

**Comment by JosephKiranBabu** [comment]
@PascalHeynmoeller This isn't really a security issue, is it? Turning global discovery on, only compromises the privacy of the connecting devices, but doesn't really put them at a security risk, since the device IDs cannot be forged by an impersonator. 
Maybe I was hasty in implementing a solution. I should go deeper into Syncthing implementation to understand what the security risks are. 

Regarding anonymity, I don't think we intended Syncthing as a service that provides anonymity. There's a story for onion share #844 for anonymous file sharing. Since we don't have any good file-sharing service like NextCloud on FreedomBox,  I wouldn't recommend reducing the usability of the only one that we have currently.

**Comment by JosephKiranBabu** [comment]
Disabling Global Discovery means that any device that is not on the local network cannot connect to the Syncthing instance. This is a serious impact on usability. Syncthing doesn't promise to be anonymity-preserving application and shouldn't be used for that purpose.

Besides, Syncthing is not a client-server application. Setting these defaults on the FreedomBox alone doesn't serve any purpose. I typically Sync one folder on 3 or 4 devices, one of them being FreedomBox. The other devices won't have these defaults. So our defaults don't matter. We better stick to the defaults provided by creators of Syncthing. 

This discussion should be moved to the Syncthing project in my opinion. FreedomBox can't do much in this regard.

**Comment by PascalHeynmoeller** [comment]
@JosephKiranBabu 
I think this compromise of the privacy of the connecting devices violates the goals of Freedombox as defined on the Website. https://freedomboxfoundation.org/goals/index.en.html
A secure default configuration of every Service which is supported in Freedombox should be the goal.
As you mentioned it's not a client-server architecture using DNS instead it uses UID's and thirdparts servers who are not under our control.
If you like we can hand over the discussion to the Mailinglist for a broader audience. I would like to read what  other people think about the issues with Syncthing and the goals of Freedombox.

**Comment by JoKeyser** [comment]
@JosephKiranBabu I agree with your change of mind; we should make very clear what Syncthing is designed to do and enable global discovery on default, just like it is on all other devices. In this case, ease-of-use trumps the goal of anonymity in my opinion. The discussion about these defaults make more sense upstream.
For anonymity, other services (within FreedomBox) seem suited better.

**Comment by JoKeyser** [comment]
So in short, this is only a documentation issue in my opinion, and @PascalHeynmoeller text in the first post should be used to inform users about it, and point to anonymous alternatives.

**Comment by JosephKiranBabu** [comment]
Please see https://github.com/freedombox/Plinth/issues/1120#issuecomment-343203222

**Comment by PascalHeynmoeller** [comment]
It would be a better approach for privacy to decentralize Syncthing by providing syncthing-discosrv on Freedombox.

https://docs.syncthing.net/users/stdiscosrv.html#setting-up
>Configuring
Note
If you are running an instance of Syncthing on the discovery server, you must either add that instance to other devices using a static address or bind the discovery server and Syncthing instances to different IP addresses.

But how can we solve this?

**Comment by JosephKiranBabu** [comment]
I'm optimistic that we might find a solution to the above problem. 
But in case we don't, maybe we can host a Syncthing discovery server on the FreedomBox Foundation's infrastructure (and maybe a Dynamic DNS server too). In that case, the user doesn't technically have to trust a third party outside of the FreedomBox community. It would still suffer from the risk of being taken down and affecting users depending on it.
